roles/kubernetes/node/templates/kubelet-config.v1beta1.yaml.j2

```diff
@@ -60,66 +60,23 @@ clusterDNS:
 - {{ dns_address }}
 {% endfor %}
 {# Node reserved CPU/memory #}
-{% if kube_reserved | bool %}
-kubeReservedCgroup: {{ kube_reserved_cgroups }}
-kubeReserved:
-{% if is_kube_master | bool %}
-  cpu: {{ kube_master_cpu_reserved }}
-  memory: {{ kube_master_memory_reserved }}
-{% if kube_master_ephemeral_storage_reserved is defined %}
-  ephemeral-storage: {{ kube_master_ephemeral_storage_reserved }}
-{% endif %}
-{% if kube_master_pid_reserved is defined %}
-  pid: "{{ kube_master_pid_reserved }}"
-{% endif %}
-{% else %}
-  cpu: {{ kube_cpu_reserved }}
-  memory: {{ kube_memory_reserved }}
-{% if kube_ephemeral_storage_reserved is defined %}
-  ephemeral-storage: {{ kube_ephemeral_storage_reserved }}
-{% endif %}
-{% if kube_pid_reserved is defined %}
-  pid: "{{ kube_pid_reserved }}"
-{% endif %}
-{% endif %}
-{% endif %}
-{% if system_reserved | bool %}
-systemReservedCgroup: {{ system_reserved_cgroups }}
-systemReserved:
-{% if is_kube_master | bool %}
-  cpu: {{ system_master_cpu_reserved }}
-  memory: {{ system_master_memory_reserved }}
-{% if system_master_ephemeral_storage_reserved is defined %}
-  ephemeral-storage: {{ system_master_ephemeral_storage_reserved }}
-{% endif %}
-{% if system_master_pid_reserved is defined %}
-  pid: "{{ system_master_pid_reserved }}"
-{% endif %}
-{% else %}
-  cpu: {{ system_cpu_reserved }}
-  memory: {{ system_memory_reserved }}
-{% if system_ephemeral_storage_reserved is defined %}
-  ephemeral-storage: {{ system_ephemeral_storage_reserved }}
-{% endif %}
-{% if system_pid_reserved is defined %}
-  pid: "{{ system_pid_reserved }}"
-{% endif %}
-{% endif %}
+{% for scope in "kube", "system" %}
+{% if lookup('ansible.builtin.vars', scope + "_reserved") | bool %}
+{{ scope }}ReservedCgroup: {{ lookup('ansible.builtin.vars', scope + '_reserved_cgroup') }}
+{{ scope }}Reserved:
+{% for resource in "cpu", "memory", "ephemeral-storage", "pid" %}
+  {{ resource }}: "{{ lookup('ansible.builtin.vars', scope + '_' ~ (resource | replace('-', '_')) + '_reserved', default='') }}"
+{% endfor %}
```

Comment:
when I read 
```
cpu: {{ kube_master_cpu_reserver}}
```
I can quickly lookup for kube_master_cpu_reserver value in the code/vars, I don't have to think

with:
```
 {{ resource }}: "{{ lookup('ansible.builtin.vars', scope + '_' ~ (resource | replace('-', '_')) + '_reserved', default='') }}"
```
I'd have to go trough all variables of the two loops before knowing what to look for.  

In templating, I prefer to write a more verbose but looking closer to rendered file, rather than a few lines of code but more abstracted(loops) and 'harder' to follow.

Reply:
Well I guess it's a matter of preference. I hadn't plan to refactor this initially (this was a preparatory step for #10714 ) but the long sequence of nested if was giving me headaches.

> I'd have to go trough all variables of the two loops before knowing what to look for.

I find that easier than trying to understand in which conditional branch you're in.

Reply:
Yes, as a developer, it's way nicer; but as a end-user trying to deploy Kubespray, it's a bit more complex to project what this file will look like and what can be configured.

Reply:
But the templates are not intended for end users. The defaults files and docs/vars.md are.